custom-metrics-stackdriver-adapter/adapter_test.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"testing"
+)
+
+var urltests = map[string]bool{
+	"https://monitoring.googleapis.com/": true, // true means that url is correct
+	"https://monitoring.googleapis.com":  true,
+	"http://monitoring.googleapis.com/":  true,
+	"http://monitoring.googleapis.com":   true,
+	"http//google.com":                   false,
+	"http//google.com/":                  false,
+	"google.com":                         false,
+	"google.com/":                        false,
+	"google/com":                         false,
+	"http:::/not.valid/a//a??a?b=&&c#hi": false,
+	"http://google.com":                  true,
+	"http://":                            false,
+}
+
+func TestURLValidator(t *testing.T) {
+	for url, correct := range urltests {
+		t.Run(url, func(t *testing.T) {
+			result := validateUrl(url)
+			if result != correct {
+				t.Errorf("for url %v got %v, want %v", url, result, correct)
+			}
+		})
+	}
+}
```

Comment:
```suggestion
func TestURLValidator(t *testing.T) {
    for _, tc := []struct{
      url string
      expect bool
    } {
     {
       url: "https://monitoring.googleapis.com/",
       expect: true
     }
    }
		t.Run(tc.url, func(t *testing.T) {
			result := validateUrl(tc.url)
			if result != tc.expect {
				t.Errorf("for url %v got %v, expect %v", tc.url, result, expect)
			}
		})
	}
}
```
Let's use table tests, 

FYI No need to pass `expect: false` as bool default is false.